cognite/client/_api/files.py

```diff
@@ -620,12 +626,53 @@ def download(
                 >>> c = CogniteClient()
                 >>> c.files.download(directory="my_directory", id=[1,2,3], external_id=["abc", "def"])
         """
+
         if isinstance(directory, str):
             directory = Path(directory)
+        assert directory.is_dir(), f"{directory} is not a directory"
+
         all_ids = IdentifierSequence.load(id, external_id).as_dicts()
         id_to_metadata = self._get_id_to_metadata_map(all_ids)
-        assert directory.is_dir(), f"{directory} is not a directory"
-        self._download_files_to_directory(directory, all_ids, id_to_metadata)
+
+        if not keep_directory_structure:
+            all_file_names = [
+                cast(str, _metadata.name) for _id, _metadata in id_to_metadata.items() if isinstance(_id, int)
+            ]
+            duplicate_names = [name for name, count in collections.Counter(all_file_names).items() if count > 1]
+            if duplicate_names:
+                duplicate_names.sort()
+                warning_message = f"""There are {len(duplicate_names)} duplicate file names.
+    Only the contents of one of the files with the same name will be downloaded to the same directory.
+    This concerns: {duplicate_names}"""
```

Comment:
Using implicit string concat. is preferred:
```suggestion
warning_message = (
                    f"There are {len(duplicate_names)} duplicate file names. Only the contents of one of the files "
                    f"with the same name will be downloaded to the same directory. This concerns: {duplicate_names}"
                )
```